create-proxy.sh

```diff
@@ -36,33 +39,17 @@ COMMAND=""
 
 if [ -e "${PSK_FILE}" ]; then
   COMMAND="chown zabbix:zabbix \"/var/lib/${PSK_FILE}\"; chmod 600 \"/var/lib/${PSK_FILE}\"; "
+fi
 if [ -e "${CA_FILE}" ]; then
   COMMAND="${COMMAND}chown zabbix:zabbix \"/var/lib/${CA_FILE}\"; chmod 600 \"/var/lib/${CA_FILE}\"; "
+fi
 if [ -e "${KEY_FILE}" ]; then
   COMMAND="${COMMAND}chown zabbix:zabbix \"/var/lib/${KEY_FILE}\"; chmod 600 \"/var/lib/${KEY_FILE}\"; "
+fi
 if [ -e "${CERT_FILE}" ]; then
   COMMAND="${COMMAND}chown zabbix:zabbix \"/var/lib/${CERT_FILE}\"; chmod 600 \"/var/lib/${CERT_FILE}\"; "
 fi
 
-COMMAND="${COMMAND}${START_CMD}"
+export COMMAND="${COMMAND}${START_CMD}"
 
-docker create \
-  --restart=unless-stopped \
-  -v ${DIR}/zabbix/odbcinst.ini:/etc/odbcinst.ini \
-  -v ${DIR}/zabbix/odbc.ini:/etc/odbc.ini \
-  -v ${DIR}/zabbix/odbc:/var/lib/zabbix/odbc \
-  -v ${DIR}/zabbix/enc:/var/lib/zabbix/enc \
-  -v ${DIR}/zabbix/externalscripts:/usr/lib/zabbix/externalscripts \
-  -v ${DIR}/zabbix/mibs:/var/lib/zabbix/mibs \
-  -v ${DIR}/zabbix/modules:/var/lib/zabbix/modules \
-  -v ${DIR}/zabbix/snmptraps:/var/lib/zabbix/snmptraps \
-  -v ${DIR}/zabbix/ssh_keys:/var/lib/zabbix/ssh_keys \
-  -v ${DIR}/zabbix/ssl/certs:/var/lib/zabbix/ssl/certs \
-  -v ${DIR}/zabbix/ssl/keys:/var/lib/zabbix/ssl/keys \
-  -v ${DIR}/zabbix/ssl/ssl_ca:/var/lib/zabbix/ssl/ssl_ca \
-  -p 10051:10051 \
-  --name ${NAME} \
-  --env-file env.list \
-  --entrypoint=/bin/bash \
-  ${CONTAINER_IMAGE}:${CONTAINER_VERSION} \
-  -c "${COMMAND}"
+docker-compose up -d
```

Comment:
I think the containers should be created but not started. Ie. `up --no-start`

Also, describe in README.md how the containers will be started and stopped (`docker-compose start` and `docker-compose stop`)

Reply:
Would that have some benefit over up/down?

Reply:
Yes, we don't want to accidentally start containers on both servers (in cluster) at the same time; otherwise Zabbix monitorings would behave incorrectly. Zabbix proxy would start polling in both servers and also Zabbix agents would send data via both proxy servers to Iiris.

Eg. when doing Zabbix proxy version upgrade on the passive server, we want to start the containers only after a successful version upgrade and when we can interchange active/passive nodes.

Reply:
That's a valid point.

Could that be a problem if server reboots since restart policy is unless-stopped (i.e. container status is not stopped but created and would start automatically after reboot)?

Reply:
Also, what is the reason to use active/passive proxy instead of active/active?

Reply:
I have believed (not tested or verified in any ways) that if container status is created (not started, not stopped) it will not start at reboot. If it does start at reboot, it will be a problem.

Reply:
In case of active/active Zabbix Proxy:
- Both proxy components would acquire the list of monitored items from Zabbix server and start JMX and HTTP polling monitorings in the customer network; data would be duplicate in Zabbix server
- Zabbix agents would actually behave nicely (contra what I said earlier) because they connect to load balancer address which is configured to forward the traffic to the primary Zabbix proxy

Reply:
Makes sense. So --no-start instead of -d and instructions updated on docker-compose start/stop. :+1: